backend/src/server/routes/v1/project-router.ts

```diff
@@ -372,6 +372,44 @@ export const registerProjectRouter = async (server: FastifyZodProvider) => {
     }
   });
 
+  server.route({
+    method: "PUT",
+    url: "/:workspaceSlug/audit-logs-retention",
+    config: {
+      rateLimit: writeLimit
+    },
+    schema: {
+      params: z.object({
+        workspaceSlug: z.string().trim()
+      }),
+      body: z.object({
+        auditLogsRetentionDays: z.number().min(1).max(100)
```

Comment:
Let keep the max even smaller. 100 is more than 3 months a bit too much. 

@maidul98 Your opinion please?

Reply:
actually, the max is dictated by the value of `auditLogsRetentionDays` in the project's subscriptio plan. We're also doing this check in the backend. What do you think? 
<img width="946" alt="image" src="https://github.com/Infisical/infisical/assets/65645666/e95fbeee-a001-4db8-8e4a-619ee261ec3d">

Reply:
i'm thinking some organizations might need more days, so to be flexible I didn't define a max in the schema but rather did a check against the subscription in the service layer instead

Reply:
Retention period to pricing is a seperate topic, we can open up a thread about it in slack @akhilmhdh . I think this setting should only really be shown to self hosted instances. On cloud, all the user could do is lower the retention days which i don't think they care to do